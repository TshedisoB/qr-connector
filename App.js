import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import { Provider } from "react-redux";

import store from "./src/Redux/store";
import ScanQr from "./src/Components/ScanQr/ScanQr";
import ContactList from "./src/Components/ContactList/ContactList";
import ContactDetails from "./src/Components/ContactDetails/ContactDetails";
import ScanButton from "./src/Components/ScanButton/ScanButton";

const Stack = createStackNavigator();

const App = () => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator
          screenOptions={{
            headerTitleAlign: "center",
          }}>
          <Stack.Screen
            name="Scan Qr Code"
            component={ScanQr}
            options={{ headerLeft: null }}
          />
          <Stack.Screen
            name="Contact List"
            component={ContactList}
            options={{ headerLeft: null }}
          />
          <Stack.Screen
            name="Contact Details"
            component={ContactDetails}
            options={{ headerLeft: null }}
          />
        </Stack.Navigator>
        <ScanButton />
      </NavigationContainer>
    </Provider>
  );
};

export default App;
