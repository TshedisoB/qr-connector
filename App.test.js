import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import App from "./App";

function MockBarCodeScanner() {
  return <></>;
}

MockBarCodeScanner.requestPermissionsAsync = async function () {
  return { status: "granted" };
};

jest.mock("expo-barcode-scanner", () => ({
  __esModule: true,
  BarCodeScanner: MockBarCodeScanner,
}));

jest.mock("./src/sqlQueries/sqlQueries", () => ({
  __esModule: true,
  createTable: jest.fn(),
  insertToDataBase: jest.fn(),
  getAllContacts: jest.fn(),
}));

describe("App component: ", () => {
  let tree;

  beforeAll(async () => {
    await act(async () => {
      tree = renderer.create(<App />);
    });
  });

  it("should render 'App component' correctly", () => {
    act(() => {
      expect(tree).toMatchSnapshot();
    });
  });

  it("should display button with text `Scan QR Code`", () => {
    const scanQRButton = tree.root.findByProps({
      testID: "scan-qr-button",
    });

    expect(scanQRButton.props.children).toEqual("Scan QR Code");
  });

  it("should display button with text `Contact List`", () => {
    const contactListButton = tree.root.findByProps({
      testID: "contact-list-button",
    });

    expect(contactListButton.props.children).toEqual("Contact List");
  });

  it("should be able to navigate to `Contact List` screen", async () => {
    const { getByTestId } = render(<App />);
    const contactListButton = getByTestId("contact-list-button");

    fireEvent.press(contactListButton);

    await waitFor(() => getByTestId("null-contacts-list"));

    expect(getByTestId("null-contacts-list")).toBeTruthy();
  });
});
