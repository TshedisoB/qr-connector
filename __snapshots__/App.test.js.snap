// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component:  should render 'App component' correctly 1`] = `
[
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "zIndex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            onLayout={[Function]}
            pointerEvents="box-none"
            style={null}
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                      "opacity": 1,
                    }
                  }
                />
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    aria-level="1"
                    collapsable={false}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Scan Qr Code
                  </Text>
                </View>
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                      "opacity": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            collapsable={false}
            forwardedRef={[Function]}
            gestureResponseDistance={
              {
                "bottom": -1,
                "end": -1,
                "start": -1,
                "top": -1,
              }
            }
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                [
                  {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                {
                  "close": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={1}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    {
                      "flex": 1,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      [
                        {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        [
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#fff",
                              "flex": 1,
                              "justifyContent": "center",
                              "left": -17,
                              "width": "110%",
                            }
                          }
                        >
                          <ViewManagerAdapter_ExpoLinearGradient
                            proxiedProperties={
                              {
                                "colors": [
                                  4286578816,
                                  4286578688,
                                  4286578816,
                                ],
                                "endPoint": undefined,
                                "locations": undefined,
                                "startPoint": undefined,
                              }
                            }
                            style={
                              {
                                "alignItems": "center",
                                "backgroundColor": "purple",
                                "borderRadius": 8,
                                "flex": 1,
                                "justifyContent": "center",
                                "overflow": "hidden",
                                "width": "100%",
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "justifyContent": "center",
                                  "marginTop": -50,
                                  "position": "relative",
                                  "width": "85%",
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "borderColor": "green",
                                    "borderStyle": "dotted",
                                    "borderWidth": 3,
                                    "height": 250,
                                    "width": 250,
                                  }
                                }
                              />
                              <View
                                style={
                                  {
                                    "bottom": 10,
                                    "height": 50,
                                    "position": "absolute",
                                  }
                                }
                              >
                                <Text
                                  style={
                                    {
                                      "alignSelf": "center",
                                      "backgroundColor": "black",
                                      "borderColor": "green",
                                      "borderRadius": 5,
                                      "borderWidth": 2,
                                      "bottom": "1.8%",
                                      "color": "green",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                      "padding": 3,
                                      "position": "absolute",
                                      "right": -90,
                                    }
                                  }
                                >
                                  Fit QR code in dotted square
                                </Text>
                              </View>
                            </View>
                          </ViewManagerAdapter_ExpoLinearGradient>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
      </View>
    </RNCSafeAreaProvider>
  </View>,
  <View
    style={
      {
        "backgroundColor": "#58b3f4",
        "flexDirection": "row",
        "height": "7%",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#0092de",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 15,
          "width": "48%",
        }
      }
      testID="scan-btn-container"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "margin": 5,
          }
        }
        testID="scan-qr-button"
      >
        Scan QR Code
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "fontSize": 25,
            },
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#0092de",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 15,
          "width": "48%",
        }
      }
      testID="contact-btn-container"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "margin": 5,
          }
        }
        testID="contact-list-button"
      >
        Contact List
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "fontSize": 25,
            },
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
